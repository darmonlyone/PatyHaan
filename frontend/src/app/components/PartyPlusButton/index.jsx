import React from "react";
import { mdiPlusThick } from "@mdi/js";

const PartyPlusButton = ({ onClick }) => {
	return (
		<button
			data-testid="plusbutton"
			onClick={onClick}
			className="fixed right-4 bottom-4 lg:right-6 lg:bottom-6 rounded-full bg-yellow-200 p-4 lg:p-6"
		>
			<svg className="text-white w-6 h-6" viewBox="0 0 24 24">
				<path d={mdiPlusThick} />
			</svg>
		</button>
	);
};

export default PartyPlusButton;
