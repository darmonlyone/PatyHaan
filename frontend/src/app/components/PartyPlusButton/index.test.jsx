import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import PartyPlusButton from ".";

const onClick = jest.fn()
it("should render PartyPlusButton with default value", async () => {
	const { asFragment } = render(
		<Router>
			<PartyPlusButton onClick={onClick } />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});

it("should render PartyPlusButton with handle click", async () => {
	const { getByTestId } = render(
		<Router>
			<PartyPlusButton onClick={onClick} />
		</Router>
	);
	
	const plusButton = getByTestId("plusbutton")
	fireEvent.click(plusButton)
	expect(plusButton).toBeDefined()
	expect(onClick).toBeCalled()
});