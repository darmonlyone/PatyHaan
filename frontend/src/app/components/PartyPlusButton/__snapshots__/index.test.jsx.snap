// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PartyPlusButton with default value 1`] = `
<DocumentFragment>
  <button
    class="fixed right-4 bottom-4 lg:right-6 lg:bottom-6 rounded-full bg-yellow-200 p-4 lg:p-6"
    data-testid="plusbutton"
  >
    <svg
      class="text-white w-6 h-6"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 14H14V20H10V14H4V10H10V4H14V10H20V14Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
